dspace-api/src/main/java/org/dspace/app/statistics/clarin/ClarinMatomoTracker.java

```diff
@@ -177,8 +174,23 @@ protected String getIpAddress(HttpServletRequest request) {
         }
         if (header != null) {
             String[] ips = header.split(", ");
-            ip = ips.length > 0 ? ips[0] : "";
+            for (String candidateIp : ips) {
+                // Validate if it's an IPv4 address
+                if (isIPv4Address(candidateIp)) {
+                    ip = candidateIp;
```

Comment:
return candidateIp?

return null at the end?

Reply:
Yes, candidateIp is a IPv4 from the `X-Forwarded-For` header. And yes it could return a null.